owner/src/main/java/org/the_chance/honeymart/ui/addCategory/AddCategoryViewModel.kt

```diff
@@ -0,0 +1,151 @@
+package org.the_chance.honeymart.ui.addCategory
+
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.flow.update
+import org.the_chance.honeymart.domain.usecase.AddCategoryUseCase
+import org.the_chance.honeymart.domain.usecase.GetAllCategoriesInMarketUseCase
+import org.the_chance.honeymart.domain.util.ErrorHandler
+import org.the_chance.honeymart.ui.base.BaseViewModel
+import javax.inject.Inject
+
+@HiltViewModel
+class AddCategoryViewModel @Inject constructor(
+    private val addToCategoryUseCase: AddCategoryUseCase,
+    private val getAllCategories: GetAllCategoriesInMarketUseCase
+) : BaseViewModel<AddCategoryUIState, AddCategoryUIEffect>(AddCategoryUIState()),
+    AddCategoryInteractionListener {
+
+    override val TAG: String = "addCategory"
+
+    init {
+        getCategoryImages()
+        getAllCategory(8)
+    }
+
+    private fun getCategoryImages() {
+        _state.update {
+            it.copy(
+                isLoading = false,
+                categoryImages = categoryImages.map { it.toCategoryImageUIState() })
+        }
+    }
+
+    private fun addCategory(name: String, CategoryImageID: Int ){
+        _state.update { it.copy(isLoading = true) }
+        tryToExecute(
+            function = { addToCategoryUseCase.invoke(name, CategoryImageID) },
+            onSuccess = ::addCategorySuccess,
+            onError = ::addCategoryError
+
+        )
+
+    }
+
+    private fun addCategorySuccess(success: String) {
+        _state.update { it.copy(isLoading = false, nameCategory = "") }
+        getAllCategory(8)
```

Comment:
No understand why need recall fun **getAllCategory()** when success,It needs clarification. You made an update when adding it.